src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -111,6 +115,23 @@ internal sealed partial class LocalRewriter : BoundTreeRewriterWithStackGuard
             }
         }
 
+        [Conditional("DEBUG")]
+        private static void AssertValidLocalRewriting(BoundNode node)
+        {
+            switch (node.Kind)
+            {
+                case BoundKind.UsingStatement:
+                case BoundKind.AwaitableValuePlaceholder:
+                    Debug.Assert(false, $"Bound nodes of kind {node.Kind} should not survive past local rewriting");
+                    break;
+            }
+
+            foreach (var child in ((IBoundNodeWithIOperationChildren)node).Children)
```

Comment:
Are we using `IOperation` elsewhere for walking `BoundNodes`? If not, consider writing a simple `BoundTreeVisitor` instead.